mindmeld/components/translators.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2015 Cisco Systems, Inc. and others.  All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module contains translator clients used by the MultiLingual Annotator.
+"""
+from abc import ABCMeta, abstractmethod
+import importlib
+import logging
+import os
+
+
+logger = logging.getLogger(__name__)
+
+
+class Translator(metaclass=ABCMeta):
+    """Abstract Translator Base Class for Translators to be used by Mindmeld."""
+
+    def __init__(self):
+        """Creates a translation client after finding the credential path."""
+        self.translate_client = None
+
+    @abstractmethod
+    def get_translate_client(self):
+        """
+        Args:
+            text (str): Input text
+        Returns:
+            language_code (str): Detected Language Code
+        """
+        raise NotImplementedError("Subclasses must implement this method")
+
+    @abstractmethod
+    def detect_language(self, text):
+        """
+        Args:
+            text (str): Input text
+        Returns:
+            language_code (str): Detected Language Code
+        """
+        raise NotImplementedError("Subclasses must implement this method")
+
+    @abstractmethod
+    def translate(self, text, target_language):
+        """
+        Args:
+            text (str): Input text
+            destination_language (str): Language code for target language.
+        Returns:
+            translated_text (str): Translated text
+        """
+        raise NotImplementedError("Subclasses must implement this method")
+
+
+class NoOpTranslator(Translator):
+    """No-Ops translator to be used when a Translator is not selected or available."""
+
+    def __init__(self):
+        pass
+
+    def get_translate_client(self):
+        return
+
+    def detect_language(self, text):
+        return
+
+    def translate(self, text, target_language):
+        return
```

Comment:
shouldnt NoOpTranslator just return the text? If it returns None, I can see the caller getting confused.

Reply:
Good point, returning text